app_dart/lib/src/request_handlers/append_log.dart

```diff
@@ -3,25 +3,37 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:typed_data';
 
 import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/service/stackdriver_logger.dart';
 import 'package:gcloud/db.dart';
 import 'package:meta/meta.dart';
 
 import '../datastore/cocoon_config.dart';
 import '../model/appengine/key_helper.dart';
 import '../model/appengine/log_chunk.dart';
+import '../model/appengine/task.dart';
 import '../request_handling/api_request_handler.dart';
 import '../request_handling/authentication.dart';
 import '../request_handling/body.dart';
 import '../request_handling/exceptions.dart';
 
 @immutable
 class AppendLog extends ApiRequestHandler<Body> {
-  const AppendLog(
+  AppendLog(
     Config config,
-    AuthenticationProvider authenticationProvider,
-  ) : super(config: config, authenticationProvider: authenticationProvider);
+    AuthenticationProvider authenticationProvider, {
+    StackdriverLoggerService stackdriverLogger,
+    @visibleForTesting Uint8List requestBodyValue,
+  })  : stackdriverLogger =
+            stackdriverLogger ?? StackdriverLoggerService(config: config),
+        super(
+            config: config,
+            authenticationProvider: authenticationProvider,
+            requestBodyValue: requestBodyValue);
+
+  final StackdriverLoggerService stackdriverLogger;
```

Comment:
I have a couple of questions/concerns regarding moving log to Stackdriver:

1. Now it is easy to search with Stackdriver, which is good. However our existing logic is also easy to search.
2. With Stackdriver, I just tried searching and loading log data
2.1 It is not convenient to loading log data, is there any way to have a one-time load for all logs? Currently I have to keep scrolling down or up to load next chunk of log. Once loaded, and I have to load it again when scroll back. 
2.2 Once I searched some key-words, and after I clicked one record in the middle, it is difficult for me to view neighborhood logs.
Both of these would make it not easy to debug/look for specific information.
3. Is Stackdriver mainly for log view of executing tasks, or both? Considering above two points, have you tried GCS? I see there is a bucket named 'flutter-task-logs' in flutter-dashboard. Could we at least point to log file in GCS for finished tasks?

Please let me know if I am wrong.

Reply:
1. This is just to let the team evaluate Stackdriver. From https://flutter.dev/go/build-dashboard-v2 the team requested a GUI for viewing logs, and this is a possible solution.
2. Scrolling to see the whole log is a pain. However, a few contributors pointed out that they only really check the end of the log as that's where the failures tend to be. Can you clarify what you mean by neighborhood logs?
3. This PR implemented GCS initially, but there was no direct advantage to it. Initially I investigated GCS as a solution as I thought Stackdriver would support that, but on implementation I was unable to load a GCS file in the Stackdriver viewer.

Reply:
2.2 Suppose we have 1000 lines (line 1 to line 1000) in the log, and searching 'error' returns me 4 lines (line 100, line 200, line 300, line 400). If one wants to see the pre or post logs around line 300, he has to clear the 'error' keyword, and refresh the page, and then scroll to the lines around 200.

Oh, there is one option to 'pin this log' at the end of each line. This seems to help.

3. GCS at least gives the quick downloading option for the whole log, if people wants to search around/move around/scroll the whole log file. 

Anyway, this is a temporary solution for log view. We can keep Stackdriver to see how it goes.